lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,250 @@
+//go:build rpctest
+// +build rpctest
+
+package itest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+	"github.com/stretchr/testify/require"
+)
+
+type chanFundMaxTestCase struct {
+	// name is the name of the target test case.
+	name string
+
+	// amount is the amount in the Carol's wallet.
+	amount btcutil.Amount
+
+	// pushAmt is the amount to be pushed to Dave.
+	pushAmt btcutil.Amount
+
+	// feeRate is an optional fee in satoshi/bytes used when opening a
+	// channel.
+	feeRate btcutil.Amount
+
+	// carolBalance is Carol's expected balance in her channel before
+	// subtracting the reserved amount to pay for a commitment transaction.
+	carolBalance btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail.
+	chanOpenShouldFail bool
+}
+
+// testChannelFundMax checks various channel funding scenarios where the user
+// instructed the wallet to use all remaining funds.
+func testChannelFundMax(net *lntest.NetworkHarness,
+	ht *harnessTest) {
+
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	carol := net.NewNode(ht.t, "Carol", nil)
+	defer shutdownAndAssert(net, ht, carol)
+
+	dave := net.NewNode(ht.t, "Dave", nil)
+	defer shutdownAndAssert(net, ht, dave)
+
+	// Ensure both sides are connected so the funding flow can be properly
+	// executed.
+	net.EnsureConnected(ht.t, carol, dave)
+
+	var testCases = []*chanFundMaxTestCase{
+		{
```

Comment:
could you add a more detailed comment why 2_000 sats is dust here, whats the feerate used 1sat/vbyte? I thought dustlimit is 330 sats so `2000-fees` should still account for the dustlimit ?

Reply:
Good idea in general to make the tests more descriptive in order to make it more intelligible for new joiners.
I've changed the tests so that we now check against specific error messages which should make the test case clearer.
Here specifically, `2_000` denotes the initial wallet balance, which would leave dust in the wallet after subtracting the miner fees which I changed to 20 sats/vbyte to make it clearer.